File: go/gin/internal/modules/address/controller.go
package address

import (
	"net/http"
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/address/dto"

	"github.com/gin-gonic/gin"
)

type AddressController struct{
	service AddressService
}


func (c *AddressController) Create(ctx *gin.Context){
	auth := helper.GetAuth(ctx)

	var dto dto.CreateAddressDto
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err.Error())
		ctx.JSON(badRequest.StatusCode, badRequest)
		return
	}

	// Transformações Necessárias
	dto.Transform()
	
	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	_, err := c.service.Create(dto, auth.ID)

	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.Status(http.StatusCreated)
}

File: rust/actix/src/modules/address/controller.rs
use actix_web::{web, HttpResponse};
use validator::Validate;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::messages::error::ErrorMessages;
use super::dto::CreateAddressDto;
use super::repository::ICreateAddressData;
use super::service::AddressService;
use crate::modules::auth::interfaces::JwtPayload;
use crate::common::errors::base::BaseError;
use uuid::Uuid;

pub struct AddressController;

impl AddressController {
    pub async fn create(
        jwt_payload: web::ReqData<JwtPayload>,
        dto: web::Json<CreateAddressDto>,
    ) -> Result<HttpResponse, BaseError> {
        let mut dto = dto.into_inner();
        dto.transform();

        if let Err(err) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        let id_client = Uuid::parse_str(&jwt_payload.id).unwrap();

        let data = ICreateAddressData {
            logradouro: dto.logradouro,
            numero: dto.numero,
            bairro: dto.bairro,
            cidade: dto.cidade,
            uf: dto.uf,
            complemento: dto.complemento,
            cep: dto.cep,
            id_cliente: id_client,
        };

        match AddressService::create(data, id_client).await {
            Ok(_) => Ok(HttpResponse::Created().finish()),
            Err(err) => Err(err),
        }
    }
}